File: FrontEnd/hotel-ms-app/src/app/core/services/room.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

// --- INTERFACES ---
export interface RoomType {
  id: number;
  typeName: string;
  pricePerHour: number;
}

export interface Room {
  id: number;
  roomNumber: string;
  roomTypeId: number;
  roomType: RoomType;
}

@Injectable({
  providedIn: 'root'
})
export class RoomService {
  private apiUrl = 'https://webapplication120250904232050-b8fteqhvgfdbbde8.canadacentral-01.azurewebsites.net/api/Rooms'; // Hardcoded URL

  constructor(private http: HttpClient) { }

  getRooms(): Observable<Room[]> {
    return this.http.get<Room[]>(this.apiUrl);
  }
}

File: BACKEND/WebApplication1/WebApplication1/Models/Room.cs
namespace WebApplication1.Models
{
   
        public class Room
        {
            public int Id { get; set; }
            public string RoomNumber { get; set; }
            public int RoomTypeId { get; set; }

            // Navigation property
            public RoomType RoomType { get; set; }
        }
    
}
